base_layer/core/src/validation/block_validators.rs

```diff
@@ -158,13 +158,11 @@ fn check_sorting_and_duplicates(body: &AggregateBody) -> Result<(), ValidationEr
 
 /// This function checks that all inputs in the blocks are valid UTXO's to be spend
 fn check_inputs_are_utxos<B: BlockchainBackend>(block: &Block, db: &B) -> Result<(), ValidationError> {
-    let data = db
-        .fetch_block_accumulated_data(&block.header.prev_hash)?
-        .ok_or(ValidationError::PreviousHashNotFound)?;
+    let deleted = db.fetch_deleted_bitmap()?;
```

Comment:
This is not equivalent. 

this in the old code would have been:
`db.fetch_block_accumulated_data(tip_height).deleted`

I am not sure if we do run this for heights where `block.header.height -1` != `tip_height`

Reply:
I think we should add a check to this validator that the block being validated is next from the tip.

Reply:
I've added that invariant/rule to the validator. The current usage of this validator is to only ever add (atomically) the next block from the tip.